src/end_entity.rs

```diff
@@ -154,7 +154,7 @@ impl<'a> EndEntityCert<'a> {
     /// Checking that a certificate is valid for a given subject name should always be done with
     /// [EndEntityCert::verify_is_valid_for_subject_name].
     #[cfg(feature = "alloc")]
-    pub fn dns_names(&'a self) -> Result<impl Iterator<Item = &'a str>, Error> {
+    pub fn dns_names(&'a self) -> impl Iterator<Item = &'a str> {
```

Comment:
Maybe tweak the docstring here? "Returns a list of valid DNS names [..]"?

Reply:
Done. I rolled this change into the commit that made the method infallible.